client/client.go

```diff
@@ -35,6 +39,32 @@ var (
 	errorAddingServerCertificateToPool = errors.New("Error adding trusted server certificate to pool.")
 )
 
+var (
+	// 	                  http.Client.Timeout
+	// +--------------------------------------------------------+
+	// +--------+  +---------+  +-------+  +--------+  +--------+
+	// |  Dial  |  |   TLS   |  |Request|  |Response|  |Response|
+	// |        |  |handshake|  |       |  |headers |  |body    |
+	// +--------+  +---------+  +-------+  +--------+  +--------+
+	// +--------+  +---------+             +--------+
+	//  Dial        TLS                     Response
+	//  timeout     handshake               header
+	//  timeout                             timeout
+	//
+	//  It covers the entire exchange, from Dial (if a connection is not reused)
+	// to reading the body. This is to timeout long lasing connections.
+	defaultClientReadingTimeout = 12 * time.Hour
+	// More granular connection timeouts
+	defaultDialTimeout           = 30 * time.Second
+	defaultTLSHandshakeTimeout   = 30 * time.Second
+	defaultResponseHeaderTimeout = 60 * time.Second
+
+	// connection keepalive options
+	connectionKeepaliveTime     = 5 * time.Second
+	connectionKeepaliveInterval = 5 * time.Second
+	connectionKeepaliveProbes   = 10
```

Comment:
Ok, I see. The `defaultClientReadingTimeout` might be useful, but I think that `defaultTLSHandshakeTimeout` and `defaultResponseHeaderTimeout` are unnecessary. Even an extremely slow connection will finish those in reasonable time, and if they are really stalling, then they will be covered by the keepalive mechanism.

Not that I am against any of them. I'm just trying to make our lives easier by not providing timeout functionality we don't need. Because I assume these would require application logic to work.